snow/engine/snowman/transitive_test.go

```diff
@@ -2988,6 +2750,202 @@ func TestEngineEarlyTerminateVoterRegression(t *testing.T) {
 	require.Equal(choices.Processing, chain[2].Status())
 }
 
+// Voting for an unissued cached block that fails verification should not
+// register any dependencies.
+//
+// Full blockchain structure:
+//
+//	  Genesis
+//	 /       \
+//	0         2
+//	|         |
+//	1         3
+//
+// We first issue block 2, and then block 3 fails verification. This causes
+// block 3 to be added to the invalid blocks cache.
+//
+// We then issue block 0, issue block 1, and accept block 0.
+//
+// If we then vote for block 3, the vote should be dropped and trigger a repoll
+// which could then be used to accept block 1.
+func TestEngineRegistersInvalidVoterDependencyRegression(t *testing.T) {
```

Comment:
We could definitely simplify this test by modifying the `non-verified` cache to include `block 3`...

I often find myself struggling with this because:
1. I want to write a regression test with the broken flow.
2. I want to write concise tests.

I have essentially gone down the route of:

- I want to encode the full flow.
- I want to spend some time to write some test helpers make testing these flows easier.